src/context.rs

```diff
@@ -471,18 +474,31 @@ impl PySessionContext {
         name: Option<&str>,
         py: Python,
     ) -> PyResult<PyDataFrame> {
-        // Instantiate pyarrow Table object & convert to batches
-        let table = data.call_method0("to_batches")?;
+        let (schema, batches) =
+            if let Ok(stream_reader) = ArrowArrayStreamReader::from_pyarrow_bound(&data) {
+                // Works for any object that implements __arrow_c_stream__ in pycapsule.
+
+                let schema = stream_reader.schema().as_ref().to_owned();
+                let batches = stream_reader
+                    .collect::<std::result::Result<Vec<RecordBatch>, arrow::error::ArrowError>>()
```

Comment:
Are you intentionally using the long form here? IMO it's easier to read as 
```suggestion
                    .collect::<Result<Vec<_>, ArrowError>()
```

Reply:
It's a mistake. During my testing I had imported a different `Result` so I had to use the long form. After removing that other import I didn't fix this up.